Framework/API/src/SpectrumInfo.cpp

```diff
@@ -212,8 +210,8 @@ SpectrumInfo::getDetectorVector(const size_t index) const {
 }
 
 std::vector<size_t> SpectrumInfo::getDetectorIndices(const size_t index) const {
-  const auto &specDef =
-      m_experimentInfo.internalSpectrumInfo().spectrumDefinition(index);
+  m_experimentInfo.updateSpectrumDefinitionIfNecessary(index);
```

Comment:
Yes, lazy initialization but also the fact that detector IDs can be added at any time, so even if it was initialized once we may need to update at any point.